js/application/init.js

```diff
@@ -0,0 +1,285 @@
+        }
+      }
+      else {
+        this.reportError(new Error("main:: Config is not defined"));
+      }
+    },
+
+    reportError: function (error) {
+      // remove loading class from body
+      domClass.remove(document.body, CSS.loading);
+      domClass.add(document.body, CSS.error);
+      // an error occurred - notify the user. In this example we pull the string from the
+      // resource.js file located in the nls folder because we've set the application up
+      // for localization. If you don't need to support multiple languages you can hardcode the
+      // strings here and comment out the call in index.html to get the localization strings.
+      // set message
+      var node = dom.byId("loading_message");
+      if (node) {
+        // Does the app want to handle the report?
+        if (this.main.reportError) {
+          this.main.reportError(node, error);
+        } else {
+          node.innerHTML = "<h1><span class=\"" + CSS.errorIcon + "\"></span> " + i18n.error + "</h1><p>" + error.message + "</p>";
+        }
+      }
+      return error;
+    },
+
+    //--------------------------------------------------------------------------
+    //
+    //  Private Methods
+    //
+    //--------------------------------------------------------------------------
+
+    _setDirection: function () {
+      var direction = this.direction;
+      var dirNode = document.getElementsByTagName("html")[0];
+      domAttr.set(dirNode, "dir", direction);
+    },
+
+    _createWebmap: function (webmapItem) {
+      var webmap;
+
+      if (!webmapItem) {
+        var error = new Error("main:: webmap data does not exist.");
+        this.reportError(error);
+        return;
+      }
+
+      if (webmapItem.data) {
+        if (webmapItem.data instanceof Error) {
+          this.reportError(webmapItem.data);
+        }
+        else {
+          webmap = new WebMap({
+            portalItem: webmapItem.data
+          });
+        }
+      }
+      else if (webmapItem.json) {
+        webmap = WebMap.fromJSON(webmapItem.json.itemData);
+        webmap.portalItem = webmapItem.json.item;
+      }
+      if (webmap) {
+        var viewProperties = {
+          map: webmap,
+          container: "viewDiv"
+        };
+        if (!this.config.title && webmap.portalItem && webmap.portalItem.title) {
+          this.config.title = webmap.portalItem.title;
+        }
+        lang.mixin(viewProperties, this.urlParamHelper.getViewProperties(this.config));
+
+        var view = new MapView(viewProperties);
+        view.then(function (response) {
+          this.urlParamHelper.addToView(view, this.config);
+
+          domClass.remove(document.body, CSS.loading);
+          document.title = this.config.title;
+
+          // Launch the app specifics
+          this.main.start({
+            config: this.config,
+            viewNode: dom.byId("viewDiv"),
+            view: view,
+            mapOrScene: webmap
```

Comment:
@MikeTschudi it just seems very odd to have a property called mapOrScene. Just have the property be what it is.